src/test/java/org/example/FilteringSearching.java

```diff
@@ -1,5 +1,147 @@
+    private final By featuredBrands = By.xpath("//span[text()='Featured Brands' or text()='Brands']/../..//span[@class='a-size-base a-color-base' and not(ancestor::div[@class='a-row a-expander-container a-expander-extend-container'])]");
+    private final By searchResults = By.xpath("//span[@data-component-type='s-search-results' and not(contains(@class, 'AdHolder'))]//span[@class='a-size-base-plus a-color-base a-text-normal' or @class='a-size-medium a-color-base a-text-normal']");
+    private final By minField = By.id("low-price");
+    private final By maxField = By.id("high-price");
+    private final By goBtn = By.xpath("//input[@class='a-button-input']");
+    private final By priceResults = By.xpath("//div[@data-component-type='s-search-result']//span[@data-a-color='base']/span[@class='a-offscreen']");
+    private final By checkedCheckbox = By.xpath("//input[@checked]/../../following-sibling::span");
+    private final By sortByDropDown = By.xpath("//span[@data-csa-c-func-deps='aui-da-a-dropdown-button']");
+    private final By lowToHigh = By.id("s-result-sort-select_1");
+    private final By highToLow = By.id("s-result-sort-select_2");
+
+
+    @BeforeClass
+    private void setDriver() {
+        System.setProperty("webdriver.chrome.driver", "src/test/resources/webdriver/chromedriver.exe");
+        driver = new ChromeDriver();
+        driver.manage().window().maximize();
+    }
+
+    @AfterClass
+    private void closeDriver() {
+        driver.quit();
+    }
+
+    @Test
+    public void testSearchingBrandedItems() {
+        openPage(AMAZON_URL);
+        clickRandomElement(categories);
+        clickRandomElement(featuredBrands);
+        Assert.assertTrue(verifyMostResultsContainBrandName());
+    }
+
+    @Test
+    private void testSearchingWithinPriceRange() {
+        openPage(AMAZON_URL);
+        clickRandomElement(categories);
+        clickRandomElement(featuredBrands);
+        int minValue = new Random().nextInt(10);
+        int maxValue = new Random().nextInt(minValue, 200);
+        filterResultsByPriceRange(minValue, maxValue);
+        Assert.assertTrue(verifyMostPricesAreWithinRange(minValue, maxValue));
+    }
+
+    @Test
+    private void testSortingResultsByPrice() {
+        openPage(AMAZON_URL);
+        clickRandomElement(categories);
+        clickRandomElement(featuredBrands);
+        click(sortByDropDown);
+        click(lowToHigh);
+        Assert.assertTrue(verifyIfCorrectlySorted(Comparator.naturalOrder()));
+        click(sortByDropDown);
+        click(highToLow);
+        Assert.assertTrue(verifyIfCorrectlySorted(Comparator.reverseOrder()));
+    }
+
+    private boolean verifyMostResultsContainBrandName() {
+        String brandName = getElement(checkedCheckbox).getText().toLowerCase();
+        List<WebElement> results = getElements(searchResults);
```

Comment:
I would add here and in the method below a size check on results list to ensure that we actually have elements in our list that we can evaluate and, if not, throw an assertion error/exception

Reply:
Added result list size check.